src/patches/lambda.js

```diff
@@ -1,8 +1,15 @@
+let xraySdk;
+try {
+  xraySdk = require('aws-xray-sdk-core');
```

Comment:
This doesn't work ('module not found') due to the way the patch is injected.

If I can figure out a way to safely require this and fetch the current segment down below, then I could make this a generally re-usable change that would benefit a lot of other services/teams. I pinged @mdlavin asking about this - I'm pretty sure no one has solved this issue yet (all trace data dropped on timeouts, including finished subsegments).

Reply:
Alternatively, this might could just be a `lambda-runtime-tools` xray helper function that wraps the handler. That might be easier to reason about and cause less surprises.